espymetrics/dal/sqlite_dal.py

```diff
@@ -13,25 +13,116 @@
+
+import boltons
 
-_MISSING = object()
 SEP = '$'  # '$' is valid in sqlite column names, no escaping required
 
+TABLE_NAME = 'on_import_data'
+CREATE_TABLE = ('CREATE TABLE IF NOT EXISTS ' + TABLE_NAME +
+                '(id integer primary key asc, ' +
+                'hostfqdn, hostname, python$argv, python$bin, ' +
+                'python$build_date, python$compiler, python$have_readline, ' +
+                'python$have_ucs4, python$is_64bit, python$version, python$version_full, ' +
+                'time$utc_epoch real, time$std_utc_offset real, ' +
+                'uname$0, uname$1, uname$2, uname$3, uname$4, uname$5, ' +
+                'username, uuid)')
+
+total_count = 0
+
+sqlite3.enable_callback_tracebacks(True)
+
 
 class SQLiteDAL(object):
     _extension = '.db'
 
+    __file_path = None
+
     def __init__(self, file_path):
-        self.file_path = file_path
+        self.__file_path = file_path
+
+        conn = sqlite3.connect(file_path)
+        conn.execute(CREATE_TABLE)
 
     def add_record(self, in_dict):
-        pass
+        query = ('INSERT INTO ' + TABLE_NAME +
+                 ' (hostfqdn, hostname, python$argv, python$bin, ' +
+                 'python$build_date, python$compiler, python$have_readline, ' +
+                 'python$have_ucs4, python$is_64bit, python$version, python$version_full, ' +
+                 'time$utc_epoch, time$std_utc_offset, ' +
+                 'uname$0, uname$1, uname$2, uname$3, uname$4, uname$5, ' +
+                 'username, uuid) VALUES (')
+        query += '"' + in_dict['hostfqdn'] + '", '
+        query += '"' + in_dict['hostname'] + '", '
+        query += '"' + in_dict['python']['argv'] + '", '
+        query += '"' + in_dict['python']['bin'] + '", '
+        query += '"' + in_dict['python']['build_date'] + '", '
+        query += '"' + in_dict['python']['compiler'] + '", '
+        query += '"' + str(in_dict['python']['have_readline'])[0] + '", '
+        query += '"' + str(in_dict['python']['have_ucs4'])[0] + '", '
+        query += '"' + str(in_dict['python']['is_64bit'])[0] + '", '
+        query += '"' + in_dict['python']['version'] + '", '
+        query += '"' + in_dict['python']['version_full'] + '", '
+        query += '"' + str(in_dict['time']['utc_epoch']) + '", '
+        query += '"' + str(in_dict['time']['std_utc_offset']) + '", '
+        query += '"' + in_dict['uname'][0] + '", '
+        query += '"' + in_dict['uname'][1] + '", '
+        query += '"' + in_dict['uname'][2] + '", '
+        query += '"' + in_dict['uname'][3] + '", '
+        query += '"' + in_dict['uname'][4] + '", '
+        query += '"' + in_dict['uname'][5] + '", '
+        query += '"' + in_dict['username'] + '", '
+        query += '"' + in_dict['uuid']
+        query += '")'
+
+        try:
+            conn = sqlite3.connect(self.__file_path)
+            conn.isolation_level = None  # autocommit
+            conn.execute(query)
+            conn.close()
+        except:
```

Comment:
This catch-all except clause is a major antipattern. Ignoring all exceptions is unpythonic. It's ok to pause a second and be specific about which exceptions you want to handle.

This block of code is not best-effort. This method should allow any exceptions raised here to propagate. The calling function can catch exceptions if it wants.

Furthermore, even though it's only four lines, so many different things happen in this try block, that I would also call this try block too large.